SYM-699: Upgrade to JakartaEE 10 (Wildfly 36.0.1)

ann-hav opened the pull request:
Note: Main commit (eb337c66) contains some code quality details not directly related to the upgrade, notably moving misplaced static methods into a couple of utility classes (`util\CalculationUtil.java`, `util\MetaDataUtil.java`).
This could be done more thoroughly, as there seems to be a number of instance methods that could (should) be made static, for performance and consistency. But since it's outside the scope of the case I chose to exclude that thoroughness from this PR. 

Consequently, in the aforementioned respect, the commit only deals with methods that were _already_ declared static on the injectable EJB classes, to address the runtime warnings emitted in the deployment phase

(A few minor syntactical fixes slipped in as well.)

ann-hav committed:
SYM-699: JakartaEE upgrade to v10

Commit contains some code quality details not directly related to the upgrade, notably moving misplaced static methods into a couple of utility classes (`util\CalculationUtil.java`, `util\MetaDataUtil.java`).
This could quite probably be done more thoroughly, as there seems to be a number of instance methods that could (should) be made static, for performance and consistency.
The commit only deals with methods that were _already_ declared static on the injectable EJB classes to address the runtime warnings emitted on deployment, since it's inappropriate for beans.

There are a few tiny syntactical fixes as well.

ann-hav committed:
SYM-699: Leaner build config

Scoping and versioning dependencies as provided **specifically** based on Wildfly 36.0.1 base layer provisions
https://www.wildfly.org/downloads/#36.0.1.Final

ann-hav committed:
SYM-699: Stability tweaks

Note: "float" without bitsize (eg "float8") in Postgres is by legacy equivalent to double type in Java.

ann-hav committed:
Update README

fritzsonhav reviewed (approved):
looks good

hav-maande reviewed (approved):
Looks good as far as I can tell

johan-ek-HaV reviewed (approved):
Looks good to me

hav-petjoh reviewed (approved):
There are several outdated plugins and dependencies that should be updated. Additionally, the following dependencies can likely be removed, as they are already included in jakartaee-api: jakarta.json-api, jakarta.ws.rs-api, and jakarta.xml.bind-api.

There are also multiple CVEs with a severity score above 10 that need to be addressed. It may be beneficial to enforce Maven and bytecode checks as well.

Otherwise, the PR looks good.

ann-hav committed:
SYM-699: Dependency clean-up

Build hygiene commit following code review #181.

Co-Authored-By: Peter Johansson <107037160+hav-petjoh@users.noreply.github.com>

ann-hav committed:
Annotate coupling for correct persistance

Annotate two-way coupling, specifically to allow deletion of `CalculationResult` entity. Problem revealed during testing after major changes to dependencies. Unclear if it's related to PR #181, however it's possibly to do with the Hibernate version bump (5->6) in eb337c66.

ann-hav committed:
Remove unused property

ann-hav committed:
Update attributions.ts
